flink-runtime/src/main/java/org/apache/flink/runtime/state/metrics/LatencyTrackingMapState.java

```diff
@@ -0,0 +1,255 @@
+                    () -> original.get(key), latencyTrackingStateMetric::updateGetLatency);
+        } else {
+            return original.get(key);
+        }
+    }
+
+    @Override
+    public void put(UK key, UV value) throws Exception {
+        if (latencyTrackingStateMetric.checkPutCounter()) {
+            trackLatencyWithException(
+                    () -> original.put(key, value), latencyTrackingStateMetric::updatePutLatency);
+        } else {
+            original.put(key, value);
+        }
+    }
+
+    @Override
+    public void putAll(Map<UK, UV> map) throws Exception {
+        if (latencyTrackingStateMetric.checkPuAllCounter()) {
+            trackLatencyWithException(
+                    () -> original.putAll(map), latencyTrackingStateMetric::updatePutAllLatency);
+        } else {
+            original.putAll(map);
+        }
+    }
+
+    @Override
+    public void remove(UK key) throws Exception {
+        if (latencyTrackingStateMetric.checkRemoveCounter()) {
+            trackLatencyWithException(
+                    () -> original.remove(key), latencyTrackingStateMetric::updateRemoveLatency);
+        } else {
+            original.remove(key);
+        }
+    }
+
+    @Override
+    public boolean contains(UK key) throws Exception {
+        if (latencyTrackingStateMetric.checkContainsCounter()) {
+            return trackLatencyWithException(
+                    () -> original.contains(key),
+                    latencyTrackingStateMetric::updateContainsLatency);
+        } else {
+            return original.contains(key);
+        }
+    }
+
+    @Override
+    public Iterable<Map.Entry<UK, UV>> entries() throws Exception {
+        if (latencyTrackingStateMetric.checkEntriesCounter()) {
+            return trackLatencyWithException(
+                    () -> original.entries(), latencyTrackingStateMetric::updateEntriesLatency);
+        } else {
+            return original.entries();
+        }
+    }
+
+    @Override
+    public Iterable<UK> keys() throws Exception {
+        if (latencyTrackingStateMetric.checkKeysCounter()) {
+            return trackLatencyWithException(
+                    () -> original.keys(), latencyTrackingStateMetric::updateKeysLatency);
+        } else {
+            return original.keys();
+        }
+    }
+
+    @Override
+    public Iterable<UV> values() throws Exception {
+        if (latencyTrackingStateMetric.checkValuesCounter()) {
+            return trackLatencyWithException(
+                    () -> original.values(), latencyTrackingStateMetric::updateValuesLatency);
+        } else {
+            return original.values();
+        }
+    }
+
+    @Override
+    public Iterator<Map.Entry<UK, UV>> iterator() throws Exception {
+        if (latencyTrackingStateMetric.checkIteratorCounter()) {
+            return trackLatencyWithException(
+                    () -> original.iterator(), latencyTrackingStateMetric::updateIteratorLatency);
```

Comment:
I think tracking the iterator creation doesn't make much sense - it's often just creation of an object (at least rocksdb and heap).

Tracking the iteration itself would make more sense. This could be done by wrapping `next()` call.
But I guess this can also be implemented later.

ditto: other iterator/iterables and other states

Reply:
Actually I planed to add such iterator wrapper to track latency.